src/main/java/com/formulaoneapi/model/Track.java

```diff
@@ -1,17 +1,40 @@
 package com.formulaoneapi.model;
 
+import com.formulaoneapi.validation.ContinentExists;
+import com.formulaoneapi.validation.CountryExists;
+import com.formulaoneapi.validation.groups.IdValidation;
 import lombok.Data;
 
 import javax.persistence.Entity;
 import javax.persistence.Id;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
 
 @Entity
 @Data
 public class Track {
     @Id
+    @NotBlank(groups = {IdValidation.class}, message = "Name cannot be blank")
+    @Size(
+            max = 255,
+            message = "Track name must not exceed 255 characters",
+            groups = {IdValidation.class} )
     private String trackName;
 
+    @CountryExists(message = "Country does not exists or is misspelled")
+    @NotBlank(groups = {IdValidation.class}, message = "Localisation cannot be blank")
+    @Size(
+            max = 56,
+            message = "Localisation must not exceed 56 characters",
+            groups = {IdValidation.class} )
```

Comment:
Is there a reason for adding @Size validations to IdValidation.class? My idea was to use this group for id because we don't pass id in body in PUT. I know that in your PUT this parameter is still validated by `@CountryExists` so it will not validate string greater than 56 but adding this to IdValidation.class is slightly misleading. Maybe create a new group if you need it?

Reply:
Sorry, copy paste issue XD